cognite/neat/workflows/steps/lib/cdf_resources.py

```diff
@@ -0,0 +1,262 @@
+import logging
+import time
+from typing import Tuple
+from cognite.neat.graph.loaders import upload_labels
+from cognite.neat.graph.loaders.core.rdf_to_assets import (
+    NeatMetadataKeys,
+    categorize_assets,
+    rdf2assets,
+    remove_non_existing_labels,
+    unique_asset_labels,
+    upload_assets,
+)
+from cognite.neat.graph.loaders.core.rdf_to_relationships import (
+    categorize_relationships,
+    rdf2relationships,
+    upload_relationships,
+)
+from cognite.neat.graph.loaders.validator import validate_asset_hierarchy
+from cognite.neat.workflows.model import FlowMessage, WorkflowConfigs
+from cognite.neat.workflows.steps.step_model import Step
+from cognite.client.data_classes import AssetFilter
+
+from cognite.client import CogniteClient
+from ..data_contracts import CategorizedAssets, CategorizedRelationships, RulesData, SolutionGraph
+
+# with contextlib.suppress(ValueError):
+#     prom_cdf_resource_stats = Gauge(
+#         "neat_graph_to_asset_hierarchy_wf_cdf_resource_stats",
+#         "CDF resource stats before and after running fast_graph workflow",
+#         ["resource_type", "state"],
+#     )
+# with contextlib.suppress(ValueError):
+#     prom_data_issues_stats = Gauge("neat_graph_to_asset_hierarchy_wf_data_issues", "Data validation issues", ["type"])
+
+
+__all__ = [
+    "CreateCDFLabels",
+    "GenerateCDFAssetsFromGraph",
+    "GenerateCDFRelationshipsFromGraph",
+    "UploadCDFAssets",
+    "UploadCDFRelationships",
+]
+
+
+class CreateCDFLabels(Step):
+    description = "The step creates default NEAT labels in CDF"
+    category = "cdf_resources"
+
+    def run(self, rules: RulesData, cdf_client: CogniteClient) -> None:
+        upload_labels(cdf_client, rules.rules, extra_labels=["non-historic", "historic"])
+
+
+class GenerateCDFAssetsFromGraph(Step):
+    description = (
+        "The step generates assets from the graph ,categorizes them and stores them in CategorizedAssets object"
+    )
+    category = "cdf_resources"
+
+    def run(
+        self, rules: RulesData, cdf_client: CogniteClient, solution_graph: SolutionGraph, configs: WorkflowConfigs
+    ) -> Tuple[FlowMessage, CategorizedAssets]:
+        self.meta_keys = NeatMetadataKeys.load(
```

Comment:
Run should not set state of the step. I realize this is from copy pasting, but lets remove it. 
```suggestion
        meta_keys = NeatMetadataKeys.load(
```